plugins/PagePerformance/Archiver.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Plugins\PagePerformance;
+
+use Piwik\Plugins\PagePerformance\Columns\TimeDomCompletion;
+use Piwik\Plugins\PagePerformance\Columns\TimeDomProcessing;
+use Piwik\Plugins\PagePerformance\Columns\TimeLatency;
+use Piwik\Plugins\PagePerformance\Columns\TimeOnLoad;
+use Piwik\Plugins\PagePerformance\Columns\TimeTransfer;
+use Piwik\Tracker\Action;
+
+/**
+ * Class Archiver
+ */
+class Archiver extends \Piwik\Plugin\Archiver
+{
+    const PAGEPERFORMANCE_TOTAL_LATENCY_TIME = 'PagePerformance_latency_time';
+    const PAGEPERFORMANCE_TOTAL_LATENCY_HITS = 'PagePerformance_latency_hits';
+    const PAGEPERFORMANCE_TOTAL_TRANSFER_TIME = 'PagePerformance_transfer_time';
+    const PAGEPERFORMANCE_TOTAL_TRANSFER_HITS = 'PagePerformance_transfer_hits';
+    const PAGEPERFORMANCE_TOTAL_DOMPROCESSING_TIME = 'PagePerformance_domprocessing_time';
+    const PAGEPERFORMANCE_TOTAL_DOMPROCESSING_HITS = 'PagePerformance_domprocessing_hits';
+    const PAGEPERFORMANCE_TOTAL_DOMCOMPLETION_TIME = 'PagePerformance_domcompletion_time';
+    const PAGEPERFORMANCE_TOTAL_DOMCOMPLETION_HITS = 'PagePerformance_domcompletion_hits';
+    const PAGEPERFORMANCE_TOTAL_ONLOAD_TIME = 'PagePerformance_onload_time';
+    const PAGEPERFORMANCE_TOTAL_ONLOAD_HITS = 'PagePerformance_onload_hits';
+    const PAGEPERFORMANCE_TOTAL_PAGE_LOAD_TIME = 'PagePerformance_pageload_time';
+    const PAGEPERFORMANCE_TOTAL_PAGE_LOAD_HITS = 'PagePerformance_pageload_hits';
+
+    public function aggregateDayReport()
+    {
+        $selects = [];
+        $table  = 'log_link_visit_action';
+
+        $performanceDimensions = [
+            new TimeLatency(),
+            new TimeTransfer(),
+            new TimeDomProcessing(),
+            new TimeDomCompletion(),
+            new TimeOnLoad()
+        ];
+
+        foreach($performanceDimensions as $dimension) {
+            $column = $dimension->getColumnName();
+            $selects[] = "sum($table.$column) as {$column}_total";
+            $selects[] = "sum(                    
+                            case when $table.$column is null
+                                then 0
+                                else 1
+                            end
+                          ) as {$column}_hits";
+        }
+
+        $joinLogActionOnColumn = array('idaction_url');
+
+        $query = $this->getLogAggregator()->queryActionsByDimension([], 'log_action1.type = ' . Action::TYPE_PAGE_URL, $selects, false, null, $joinLogActionOnColumn);
```

Comment:
Instead of having a separate query, would it be possible to hook into `Actions.Archiving.addActionMetrics` event just like Bandwidth does? Would safe an additional query and maybe this way would also work for page titles automatically? We're seeing not always do people have URLs (eg when mobile app etc)

Reply:
That is actually done and it should already work for page titles.... The archiving here are the overall numbers. It's all exactly the same way as Bandwidth plugin does 🤔 

https://github.com/matomo-org/matomo/blob/2f557a0f8479ca7bb16304dbdc0f324073aef5eb/plugins/PagePerformance/PagePerformance.php#L42
https://github.com/matomo-org/matomo/blob/2f557a0f8479ca7bb16304dbdc0f324073aef5eb/plugins/PagePerformance/PagePerformance.php#L128-L131
https://github.com/matomo-org/matomo/blob/2f557a0f8479ca7bb16304dbdc0f324073aef5eb/plugins/PagePerformance/Metrics.php#L81-L124

Reply:
Not sure I understand why the extra query is needed then? Is this just to get the overall metrics? Could in that case the sum be already done in the query? Wonder if even a custom query somehow would need to be used since it wouldn't include page views with no page url?

Reply:
Might be ok though if they aren't included

Reply:
Actually guess we could simply sum up all values regardless of the action type, as the values will only be available for page views, so no need to limit it with the where condition. Will change that